### PR TITLE
feat: implement webhook_url_file for discord and msteams

### LINKED ISSUE #3536: [Feature] Add `webhook_url_file` to Teams integration
I'm super happy to have the native MS Teams integration, but git leaks flags the webhook url. Would it be possible to add a `webhook_url_file` like other integrations have?

### PR DESCRIPTION
**Pull request by tamcore**
Closes #3536 and #3482.

### CODE REVIEW

**Comment by dbluxo** [comment]
We would love to have this feature in the next release, @simonpasquier / @gotjosh could you maybe take a look?

File: notify/discord/discord.go
Comment (grobinson-grafana) [inline notify/discord/discord.go:117]: A bit of a nit, but perhaps call this `b` instead of `content`. Two reasons:

1. It's short.
2. It's a somewhat idiomatic way of telling other maintainers that this variable is a byte slice instead of a string without having to check the docs for `os.ReadFile`.

```suggestion
		b, err := os.ReadFile(n.conf.WebhookURLFile)
```

File: notify/discord/discord_test.go
Comment (grobinson-grafana) [inline notify/discord/discord_test.go:156]: Good test, to check for removing whitespace! 👍

File: notify/msteams/msteams.go
Comment (grobinson-grafana) [inline notify/msteams/msteams.go:118]:
```suggestion
		b, err := os.ReadFile(n.conf.WebhookURLFile)
```

**Review by grobinson-grafana — COMMENTED** [review]
I'm not a Alertmanager maintainer, so I cannot merge this - but in general LGTM! Nice work! 👍

**Comment by dbluxo** [comment]
@tamcore Do you have time to implement the proposed changes?

**Comment by dbluxo** [comment]
@tamcore Thanks for your response, I think being consistent with the other receivers is a valid approach.

**Review by gotjosh — APPROVED** [review]
LGTM

Thank you so much for your contribution this looks great - @grobinson-grafana comments are very valid, but as you said this aligns with other notifiers so I'm good with this.

**Pull request merged**
